src/v/model/record.h

```diff
@@ -647,16 +670,32 @@ class record_batch {
     const iobuf& data() const { return _records; }
     iobuf&& release_data() && { return std::move(_records); }
     void clear_data() { _records.clear(); }
+    void key_offsets(std::vector<key_offset>&& key_offsets) {
+        _key_offsets = ss::make_lw_shared<std::vector<key_offset>>(
+          std::move(key_offsets));
+    }
+    const ss::lw_shared_ptr<std::vector<key_offset>> key_offsets() const {
```

Comment:
this method can return `const std::vector<key_offset>&`

Reply:
do you mean `const std::vector<key_offset>& key_offsets() const { return *_key_offsets; }`?

Reply:
+1 (if we know they are always there)